src/main/java/seedu/linkedout/model/applicant/SortComparator.java

```diff
@@ -0,0 +1,76 @@
+package seedu.linkedout.model.applicant;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+
+public class SortComparator implements Comparator<Applicant> {
+
+    private final Field field;
+    private final Order order;
+
+    /**
+     *
+     * @param field
+     * @param order
+     */
+    public SortComparator(Field field, Order order) {
+        // assume both pass parseUtil before passing here
+        requireNonNull(field);
+        requireNonNull(order);
+        this.field = field;
+        this.order = order;
+    }
+
+    @Override
+    public int compare(Applicant applicant1, Applicant applicant2) {
+        requireNonNull(applicant1);
+        requireNonNull(applicant2);
+        String orderString = this.order.toString();
+        String fieldString = this.field.toString();
+        switch(orderString) {
+        case "ASC":
+            switch (fieldString) {
+            case "NAME":
+                return applicant1.compareNames(applicant2);
+            case "JOB":
+                return applicant1.compareJobs(applicant2);
+            default:
+                return 0;
+            }
+        case "DESC":
+            switch (fieldString) {
+            case "NAME":
+                return applicant2.compareNames(applicant1);
```

Comment:
I believe you do not need to use compareNames method to reverse the sorted order, instead, comparator return values can be used right?

E.g 
`if (applicant1.getName().compareTo(applicant2.getName()) >= 1) {
    return -1
}`